cmd/instance_pool_create.go

```diff
@@ -61,91 +61,126 @@ func (c *instancePoolCreateCmd) cmdPreRun(cmd *cobra.Command, args []string) err
 }
 
 func (c *instancePoolCreateCmd) cmdRun(_ *cobra.Command, _ []string) error {
-	instancePool := &egoscale.InstancePool{
-		Description:    utils.NonEmptyStringPtr(c.Description),
-		DiskSize:       &c.DiskSize,
-		IPv6Enabled:    &c.IPv6,
-		InstancePrefix: utils.NonEmptyStringPtr(c.InstancePrefix),
-		Labels: func() (v *map[string]string) {
-			if len(c.Labels) > 0 {
-				return &c.Labels
-			}
-			return
-		}(),
-		Name:   &c.Name,
-		SSHKey: utils.NonEmptyStringPtr(c.SSHKey),
-		Size:   &c.Size,
+
+	ctx := gContext
+	client, err := switchClientZoneV3(ctx, globalstate.EgoscaleV3Client, c.Zone)
+	if err != nil {
+		return err
 	}
 
-	ctx := exoapi.WithEndpoint(gContext, exoapi.NewReqEndpoint(account.CurrentAccount.Environment, c.Zone))
+	sshKey := &v3.SSHKey{Name: c.SSHKey}
+
+	instancePoolReq := v3.CreateInstancePoolRequest{
+		Description:    c.Description,
+		DiskSize:       c.DiskSize,
+		Ipv6Enabled:    &c.IPv6,
+		InstancePrefix: c.InstancePrefix,
+		Labels:         c.Labels,
+		MinAvailable:   c.MinAvailable,
+		Name:           c.Name,
+		SSHKey:         sshKey,
+		Size:           c.Size,
+	}
 
 	if l := len(c.AntiAffinityGroups); l > 0 {
-		antiAffinityGroupIDs := make([]string, l)
+		instancePoolReq.AntiAffinityGroups = make([]v3.AntiAffinityGroup, l)
+		af, err := client.ListAntiAffinityGroups(ctx)
+		if err != nil {
+			return fmt.Errorf("error listing Anti-Affinity Group: %w", err)
+		}
 		for i := range c.AntiAffinityGroups {
-			antiAffinityGroup, err := globalstate.EgoscaleClient.FindAntiAffinityGroup(ctx, c.Zone, c.AntiAffinityGroups[i])
+			antiAffinityGroup, err := af.FindAntiAffinityGroup(c.AntiAffinityGroups[i])
 			if err != nil {
 				return fmt.Errorf("error retrieving Anti-Affinity Group: %w", err)
 			}
-			antiAffinityGroupIDs[i] = *antiAffinityGroup.ID
+			instancePoolReq.AntiAffinityGroups[i] = v3.AntiAffinityGroup{ID: antiAffinityGroup.ID}
 		}
-		instancePool.AntiAffinityGroupIDs = &antiAffinityGroupIDs
 	}
 
 	if c.DeployTarget != "" {
-		deployTarget, err := globalstate.EgoscaleClient.FindDeployTarget(ctx, c.Zone, c.DeployTarget)
+		targets, err := client.ListDeployTargets(ctx)
+		if err != nil {
+			return fmt.Errorf("error listing Deploy Target: %w", err)
+		}
+		deployTarget, err := targets.FindDeployTarget(c.DeployTarget)
 		if err != nil {
 			return fmt.Errorf("error retrieving Deploy Target: %w", err)
 		}
-		instancePool.DeployTargetID = deployTarget.ID
+		instancePoolReq.DeployTarget = &v3.DeployTarget{ID: deployTarget.ID}
 	}
 
 	if l := len(c.ElasticIPs); l > 0 {
-		elasticIPIDs := make([]string, l)
+		instancePoolReq.ElasticIPS = make([]v3.ElasticIP, l)
+		eIP, err := client.ListElasticIPS(ctx)
+		if err != nil {
+			return fmt.Errorf("error listing Elastic IP: %w", err)
+		}
 		for i := range c.ElasticIPs {
-			elasticIP, err := globalstate.EgoscaleClient.FindElasticIP(ctx, c.Zone, c.ElasticIPs[i])
+			elasticIP, err := eIP.FindElasticIP(c.ElasticIPs[i])
```

Comment:
now, you need to fix it like in the example I push in the update instance pool :)